classes/Telemetry.php

```diff
@@ -0,0 +1,353 @@
+<?php
+/*******************************************************************************
+ * Copyright (c) 2019, Code Atlantic LLC
+ ******************************************************************************/
+
+if ( ! defined( 'ABSPATH' ) ) {
+	// Exit if accessed directly
+	exit;
+}
+
+/**
+ * Our telemetry class.
+ *
+ * Handles sending usage data back to our servers for those who have opted into our telemetry.
+ *
+ * @since 1.11.0
+ */
+class PUM_Telemetry {
+
+	public static function init() {
+		add_action( 'pum_daily_scheduled_events', array( __CLASS__, 'track_check' ) );
+		if ( is_admin() && current_user_can( 'manage_options' ) ) {
+			add_filter( 'pum_alert_list', array( __CLASS__, 'optin_alert' ) );
+			add_action( 'init', array( __CLASS__, 'optin_alert_check' ) );
+		}
+	}
+
+	/**
+	 * Prepares and sends data, if it is time to do so
+	 * @since 1.11.0
+	 */
+	public static function track_check() {
+		if ( self::is_time_to_send() ) {
+			$data = self::setup_data();
+			self::send_data( $data );
+			set_transient( 'pum_tracking_last_send', true, 6 * DAY_IN_SECONDS );
```

Comment:
Transients are not reliable for this type of thing. They should only be used for data that is not required to be there or accurate. 

IE caching the results from an API request.

They can be cleared or malfunction very easily for something that we want to reliably keep sites from hitting our servers too much.

I'd either do set_option here, or look at an actual cron event, and of course once action scheduler is merged you can move it there.

Reply:
Ah, I copied exactly how it was in the system I removed in 1.10 and thought it was a useful idea so I kept it. That said, I was planning on implementing the action scheduler once it is merged into develop so I will replace this before release. Can go ahead and do it now or once action scheduler is merged.